pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -17,14 +17,33 @@ limitations under the License.
 package reconciler
 
 import (
+	"reflect"
+	"time"
+
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/klog/v2"
 )
 
 func (rc *reconciler) Run(stopCh <-chan struct{}) {
 	rc.reconstructVolumes()
 	klog.InfoS("Reconciler: start to sync state")
-	wait.Until(rc.reconcile, rc.loopSleepDuration, stopCh)
+	wait.DyanmicPeriodUntil(func() {
+		mountedVolumesOldState := rc.actualStateOfWorld.GetAllMountedVolumes()
+		rc.reconcile()
+		mountedVolumesNewState := rc.actualStateOfWorld.GetAllMountedVolumes()
+		// affect the next iteration
+		if reflect.DeepEqual(mountedVolumesOldState, mountedVolumesNewState) {
```

Comment:
Suggestion was to get some signal from pods being added or removed. Ideally some triggers on pod change (maybe container, not sure) will change the reconcile loop to shorter period for some time. 

The way it is implemented now we are getting longer delay to unmount.

Reply:
Hello @SergeyKanzhelev 

1.	“Getting a longer delay to unmount”: Is the delay occurring because the PR made changes only on the second iteration? Could you confirm, or is there another reason?
2.	Event-based: based on  CRI event?
3.	I  checked  another KEP with a similar design: [KEP-3386: Kubelet Evented PLEG](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3386-kubelet-evented-pleg/README.md). It looks like we’re moving in the same direction. Does that sound right? 

(Just want to make sure before adapting the PR)

Thank you  🙏🙏

Reply:
longer delay is because once we switched to higher timeout and then deleted pod- reconcile will take longer to clean up. This is why the proposal was to get some kind of signal from pods being added/removed and switch to lower timeout for some time after. It is a good alternative to switch to the event-based system completely as event-based will require way more design work

Reply:
Should I submit a design proposal first and then implement it once it’s confirmed?

Reply:
really depends on how big the change will be. Also SIG Storage is the right SIG to confirm the design first - perhaps you can stop by SIG Storage meeting and ask there

Reply:
will presented at the next "COSI Standup Meeting" (Thursday, September 12, 2024)